src/utils/renderers.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2021 PixieBrix, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { TemplateEngine } from "@/core";
+import Mustache from "mustache";
+import mapKeys from "lodash/mapKeys";
+
+const hyphenRegex = /-/gi;
+
+export type Renderer = (template: string, context: object) => string;
+
+let _nunjucks: any;
+let _handlebars: any;
+
+async function ensureNunjucks() {
+  if (_nunjucks == null) {
+    _nunjucks = (await import("nunjucks")).default;
+    _nunjucks.configure({ autoescape: true });
+  }
+  return _nunjucks;
+}
```

Comment:
I'd use a wrapper like https://github.com/sindresorhus/onetime or any memoizer to avoid globals and make it self-contained:
```suggestion
const ensureNunjucks = onetime(async () => {
  const {default: nunjucks} = await import("nunjucks");
  nunjucks.configure({ autoescape: true });
  return nunjucks; 
});
```

Reply:
> or any memoizer 

Good call! Your favorite library lodash 😏 also has a `once` method we can use

Reply:
> has a `once` method we can use

Great! Looking forward to rewrite it in #816 😁